### DISCUSSION TITLE
CUDA support for wav2vec2

### DISCUSSION THREAD

**Discussion by rafael844** [sepinf-inc/IPED#1362] [Q&A] [answered]
Hi, im using last snapshot so I can test remote wav2vec2, we have an Nvidia Quadro P1000, but iped cant see any CUDA, then uses the cpu to process.
Is there any iped config we should set the CUDA?
![image](https://user-images.githubusercontent.com/31928478/195158942-4fa75e40-44ed-432d-93dd-0baae828e621.png)
deviceQuery.exe recognizes it
![image](https://user-images.githubusercontent.com/31928478/195159526-9f291739-6df3-458e-ac2d-69d8c0ec4130.png)
![image](https://user-images.githubusercontent.com/31928478/195159838-fdb98aec-1e4f-4aae-b515-b1994fd2ad35.png)

**Comment by lfcnassif** [comment, marked as answer]
First you need to install your GPU drivers, CUDA 11.7 or CUDA 11.8 and pytorch into IPED's python. Default pytorch doesn't come with GPU support, you need to install it using one of these commands https://pytorch.org/get-started/locally/
Then you will need to install cuDNN

**Reply by lfcnassif** [reply]
Another thing, your GPU doesn't have enough memory to load jonatasgrosman's 1billion parameters models, you will have to load the smaller and less accuracy ones (lgris's or eddresson's).

**Reply by lfcnassif** [reply]
> First you need python 3.9

Just saw you are on Windows, just use IPED's python and install pytorch CUDA version into it.

**Reply by rafael844** [reply]
Thank you Nassif, after those changes, now its working!

**Reply by lfcnassif** [reply]
About your last error:
![image](https://user-images.githubusercontent.com/7276994/195687692-951b3522-9928-4614-8acb-9e33f83577d6.png)

Have you solved it using a more recent iped-4.1 snapshot?

**Reply by lfcnassif** [reply]
I just remembered you need to enable the audio transcription option in IPEDConfig.txt to start the service (this is not on the Wiki...). I'll commit a small change to don't need this setting.

**Reply by rafael844** [reply]
Yes, that was my fault, becouse of it I deleted the post. When I set the snapshot again I only set AudioTranscriptConfig.txt. When I set true the audio transcription in IPEDConfig.txt it worked very well. My process decreased from 3 hours (8 cores CPU) to 15 minutes (GPU Quadro P1000 with 4GB), using a low GPU.
Thank you.

**Reply by lfcnassif** [reply]
Are you using a smaller model or the large one?

**Reply by rafael844** [reply]
smaller, when I try jonatasgrosman/wav2vec2-xls-r-1b-portuguese, IPED sets model=CPU automatically.

**Reply by lfcnassif** [reply]
Right, it tries to load on GPU, if it fails (e.g. insufficient memory), it falls back to CPU.
